File: FootballExplorer/app/src/main/java/fr/a42consulting/footballexplorer/View/PlayersListActivity.java
package fr.a42consulting.footballexplorer.View;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.widget.ListView;
import android.widget.Toast;

import java.util.ArrayList;

import fr.a42consulting.footballexplorer.Model.PlayersListPlayer;
import fr.a42consulting.footballexplorer.Presenter.IPlayersListPresenter;
import fr.a42consulting.footballexplorer.Presenter.PlayersListPresenter;
import fr.a42consulting.footballexplorer.R;

public class PlayersListActivity extends AppCompatActivity implements IPlayersListView {
    public static final String TEAM_NAME = "INTENT_PARAM_TEAM_NAME";

    ListView playersList;

    IPlayersListPresenter playersListPresenter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_players_list);
        Bundle bundle = getIntent().getExtras();
        String team = "";
        if (bundle != null) {
            team = bundle.getString(TEAM_NAME);
        }

        //Init view
        playersList = findViewById(R.id.playersList);

        playersListPresenter = new PlayersListPresenter(this);
        playersListPresenter.onLoadPlayers(team);
    }

    @Override
    public void onRequestResult(ArrayList<PlayersListPlayer> players) {
        PlayersListAdapter listAdapter = new PlayersListAdapter(this, R.layout.list_player_element);
        playersList.setAdapter(listAdapter);
        listAdapter.addAll(players);
        listAdapter.notifyDataSetChanged();
        if (players.size() == 0) {
            onMessage("Aucun joueur trouvé.");
        } else {
            onMessage(players.size() + " joueurs(s) trouvé(s)");
        }
    }

    @Override
    public void onMessage(String message) {
        Toast.makeText(this, message, Toast.LENGTH_LONG).show();
    }
}

File: FootballExplorer_kotlin/app/src/main/java/fr/a42consulting/footballexplorer/view/PlayersListActivity.kt
package fr.a42consulting.footballexplorer.view

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.v7.widget.LinearLayoutManager
import android.widget.Toast

import java.util.ArrayList

import fr.a42consulting.footballexplorer.model.PlayersListPlayer
import fr.a42consulting.footballexplorer.presenter.IPlayersListPresenter
import fr.a42consulting.footballexplorer.presenter.PlayersListPresenter
import fr.a42consulting.footballexplorer.R
import fr.a42consulting.footballexplorer.view.adapter.PlayersListAdapter
import kotlinx.android.synthetic.main.activity_players_list.*

class PlayersListActivity : AppCompatActivity(), IPlayersListView {
    private lateinit var playersListPresenter: IPlayersListPresenter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_players_list)

        val team = intent.getStringExtra(TEAM_NAME)

        playersList.layoutManager = LinearLayoutManager(this)

        playersListPresenter = PlayersListPresenter(this)
        playersListPresenter.onLoadPlayers(team)
    }

    override fun onRequestResult(players: ArrayList<PlayersListPlayer>) {
        val listAdapter = PlayersListAdapter(this, players)
        playersList.adapter = listAdapter
        listAdapter.notifyDataSetChanged()
        if (players.size == 0) {
            onMessage(getString(R.string.players_found_no))
        } else {
            onMessage(players.size.toString() + getString(R.string.players_found))
        }
    }

    override fun onRequestFailed() {
        onMessage(getString(R.string.players_found_failed))
    }

    override fun onMessage(message: String) {
        Toast.makeText(this, message, Toast.LENGTH_LONG).show()
    }

    override fun onDestroy() {
        super.onDestroy()
        playersListPresenter.resetView(null)
        playersList.adapter = null
    }

    companion object {
        const val TEAM_NAME = "INTENT_PARAM_TEAM_NAME"
    }
}
